src/main/java/uk/gov/hmcts/sscs/builder/TrackYourAppealJsonBuilder.java

```diff
@@ -137,7 +134,7 @@ private static void buildEventNode(Event event, ObjectNode eventNode, CaseData c
                 break;
             case HEARING_BOOKED :
             case NEW_HEARING_BOOKED :
-                Hearing hearing = caseData.getHearings().get(0);
+                Hearing hearing = getHearing(event, caseData.getHearings());
```

Comment:
Santhosh, I believe there is a difference between HEARING_BOOKED and NEW_HEARING_BOOKED events. The frontend will display difference messages for these two events. Could you please check the TYA API code implementation.

Reply:
Sure sathish. I have to check with aaron regarding these scenarios, newhearingbooked, pasthearingbooked. I will raise another ticket for this.
This ticket is to get the hearing details based on the date match between hearing and event.